Sorry, this will take some time. You'll have to fix some pylint issues and testing the feature in a container does not work.
```
$ podman run -v $(pwd)/prometheus-webhook-snmp:/usr/bin/prometheus-webhook-snmp -v $(pwd)/prometheus-webhook-snmp.conf:/etc/prometheus-webhook-snmp.conf -v $(pwd)/cert.pem:/srv/cert.pem -v $(pwd)/key.pem:/srv/key.pem --env RUN_ARGS="--cert /srv/cert.pem --key /srv/key.pem" -p 9099:9099 prometheus-webhook-snmp:latest
```

I think dependencies to SSL libraries in RPM spec is missing, too. The PR needs some refactoring.

```
************* Module nordix-prometheus-webhook-snmp.prometheus_webhook_snmp.utils
prometheus_webhook_snmp/utils.py:189:0: C0330: Wrong hanging indentation (remove 1 space).
             'server.ssl_module': 'builtin',
            |^ (bad-continuation)
prometheus_webhook_snmp/utils.py:190:0: C0330: Wrong hanging indentation (remove 1 space).
             'server.ssl_certificate': ctx.config['cert'],
            |^ (bad-continuation)
prometheus_webhook_snmp/utils.py:191:0: C0330: Wrong hanging indentation (remove 1 space).
             'server.ssl_private_key': ctx.config['key']
            |^ (bad-continuation)
prometheus_webhook_snmp/utils.py:192:0: C0330: Wrong hanging indentation.
         })
        |^  | (bad-continuation)
prometheus_webhook_snmp/utils.py:195:0: C0303: Trailing whitespace (trailing-whitespace)
```

I also think the `--cert`  and `--key` options should be relocated to the `run`  command because they are only used in this context.

I've did some refactoring, but can not push the changes because of `ERROR: Permission to Nordix/prometheus-webhook-snmp.git denied to votdev.`. Did you have set the correct permissions when creating the PR?

--- a/README.md
+++ b/README.md
@@ -14,6 +14,11 @@ To start the receiver execute the command:
 
     $ ./prometheus-webhook-snmp run
 
+To start with HTTPS, pass the certificate and key parameters:
+
+    $ ./prometheus-webhook-snmp --cert "certificate.pem" --key "key.pem" run
+
+Note: Both '--cert' and '--key' are needed to enable HTTPS.
 # Send a test SNMP trap
 
 If you want to send a test SNMP trap, then simply execute the following command. This can be used to test your command line parameters.

--- a/prometheus-webhook-snmp
+++ b/prometheus-webhook-snmp
@@ -72,11 +72,18 @@ def cli(ctx, debug, snmp_host, snmp_port, snmp_community, snmp_retries,
 @click.option('--metrics',
               is_flag=True,
               help='Provide Prometheus metrics from this receiver.')
+@click.option('--cert',
+              help='The CA certificate for setting up HTTPS. '
+                   'Need to be used with --key. Without this option, HTTP is used.')
+@click.option('--key',
+              help='The CA private key for setting up HTTPS. Need to be used with --cert.')
 @pass_context
-def run(ctx, host, port, metrics):
+def run(ctx, host, port, metrics, cert, key):
     ctx.config['host'] = host
     ctx.config['port'] = port
     ctx.config['metrics'] = True if metrics else None
+    ctx.config['cert'] = cert
+    ctx.config['key'] = key
     ctx.config.dump()
     utils.run_http_server(ctx)
     sys.exit(0)

--- a/prometheus_webhook_snmp/utils.py
+++ b/prometheus_webhook_snmp/utils.py
@@ -184,6 +184,15 @@ def run_http_server(ctx):
     :param ctx: The application context.
     :type ctx: dict
     """
+    if ctx.config['cert'] and ctx.config['key']:
+        cherrypy.config.update({
+            'server.ssl_module': 'builtin',
+            'server.ssl_certificate': ctx.config['cert'],
+            'server.ssl_private_key': ctx.config['key']
+        })
+    elif ctx.config['cert'] or ctx.config['key']:
+        raise Exception("Both '--cert' and '--key' are needed")
+
     cherrypy.config.update({
         'environment': 'production',
         'server.socket_host': ctx.config['host'],
@@ -217,7 +226,9 @@ class Config(dict):
             'trap_default_severity': '',
             'host': '0.0.0.0',
             'port': 9099,
-            'metrics': False
+            'metrics': False,
+            'cert': '',
+            'key': ''
         }
 
     def dump(self):

